synapse/logging/context.py

```diff
@@ -42,13 +43,19 @@
     # exception.
     resource.getrusage(RUSAGE_THREAD)
 
+    def is_thread_resource_usage_supported():
```

Comment:
I suggest using a global variable rather than a function here, for speed and simplicity.

Reply:
Haha, same thing occurred to me after thinking about it a bit more... 😅